src/blog/2023/005-custom-form-fields.md

```diff
@@ -0,0 +1,196 @@
+    'key',
+    'label',
+    'description',
+    'min',
+    'max',
+    'disabled',
+    'readonly'
+  ]
+};
+```
+
+That's all! The component library and properties panel will be automatically updated with the new form component, and the renderer will take over the rendering of the form component.
+
+<div class="figure full-size">
+  <a href="https://demo.bpmn.io/form">
+    <img src="{{ assets }}/attachments/blog/2023/005-range-palette.png" alt="The component library with the custom range component">
+  </a>
+
+  <p class="caption">
+    The custom range component gets populated into the component library.
+  </p>
+</div>
+
+
+<div class="figure condensed-size">
+  <a href="https://demo.bpmn.io/form">
+    <img src="{{ assets }}/attachments/blog/2023/005-range-properties.png" alt="The properties panel with default entries for the custom range component">
+  </a>
+
+  <p class="caption">
+    Re-use existing properties entries.
+  </p>
+</div>
+
+Please find out more about [rendering extensions](https://github.com/bpmn-io/form-js-examples/tree/master/custom-components#add-a-custom-form-component-renderer) in our dedicated example. 
+
+
+## Create a properties panel extension
+
+We added the ability to extend the properties panel with custom entries. We mirrored the [proven provider architecture](https://bpmn.io/blog/posts/2022-new-properties-panel-foundation.html) of other bpmn.io properties panel libraries.
+
+Next to the `propertiesPanelEntries` we described above, you can create a properties provider that can add, remove, and modify existing entries.
+
+```javascript 
+export class CustomPropertiesProvider {
+  constructor(propertiesPanel) {
+    propertiesPanel.registerProvider(this, 500);
+  }
+
+  getGroups(field, editField) {
+
+    ...
+    return (groups) => {
+
+      if (field.type !== 'range') {
+        return groups;
+      }
+
+      const generalIdx = findGroupIdx(groups, 'general');
+
+      groups.splice(generalIdx + 1, 0, {
+        id: 'range',
+        label: 'Range',
+        entries: RangeEntries(field, editField)
+      });
+
+      return groups;
+    };
+  }
+}
+```
+
+<div class="figure full-size">
+  <a href="https://demo.bpmn.io/form">
+    <img src="{{ assets }}/attachments/blog/2023/005-range-custom-entries.png" alt="Custom properties panel entries for the range component">
+  </a>
+
+  <p class="caption">
+    Custom properties panel entries for the range component.
+  </p>
+</div>
+
+Find out more about [properties panel extensions](https://github.com/bpmn-io/form-js-examples/tree/master/custom-components#add-custom-properties-panel-entries) in our dedicated example.
+
+
+## Plugging everything together
+
+You can plug your custom modules into the different form-js components as `viewerAdditionalModules` and `editorAdditionalModules`. With the latest release, we also added the `additionalModules` option that allows you to provide extensions to both simultaneously, for example, the renderer.
```

Comment:
Should we give an example of `viewerAdditionalModules` too?

Reply:
I would have to think of a proper use case. In which case you'd like to have an extension for the viewer only, but not for the editor? In terms of a custom renderer you have to add to both anyways. 

One example that comes into my mind is if you want to have a different rendering in both components. We do that for the `text` component, where it looks different in viewer and editor. Maybe that could be something to showcase.